routes/api/contacts.js

```diff
@@ -1,32 +1,120 @@
+const Joi = require("joi");
 const router = express.Router();
+// const { v4: uuidv4 } = require("uuid");
+
+const phonePattern = /^[0-9]{9}$/; // Паттерн для 9 цифр
+
+const contactSchema = Joi.object({
+  name: Joi.string().required(),
+  email: Joi.string().required(),
+  phone: Joi.string().regex(phonePattern).required().messages({
+    "string.pattern.base": "Phone number must have 9 digits.",
+  }),
+});
 
-const contacts = [];
+const schemaUpdate = Joi.object({
+  name: Joi.string().min(3).max(20),
+  email: Joi.string().email(),
+  phone: Joi.string().regex(phonePattern),
+}).min(1);
 
-router.get("/api/contacts", async (req, res, next) => {
-  // res.json({ message: "template message" });
+router.get("/", async (req, res, next) => {
   try {
-    res.status(200).json(contacts);
+    const contacts = await contactsOperations.listContacts();
+    res.json({
+      status: "success",
+      code: 200,
+      data: {
+        result: contacts,
+      },
+    });
   } catch (error) {
-    res.status(500).json({ message: "Internal server error" });
+    next(error);
   }
 });
 
 router.get("/:contactId", async (req, res, next) => {
-  res.json({ message: "template message" });
+  try {
+    const { contactId } = req.params;
+    const result = await contactsOperations.getContactById(contactId);
+    if (!result) {
+      throw createError(404, "Not found");
+    }
+    res.json({
+      status: "success",
+      code: 200,
+      data: {
+        result,
+      },
+    });
+  } catch (error) {
+    next(error);
+  }
 });
 
 router.post("/", async (req, res, next) => {
-  res.json({ message: "template message" });
+  try {
+    const { error } = contactSchema.validate(req.body);
+    if (error) {
+      throw createError(400, "missing required name field");
+    }
+    const result = await contactsOperations.addContact(req.body);
+    res.status(201).json({
+      status: "success",
+      code: 201,
+      data: {
+        result,
+      },
+    });
+  } catch (error) {
+    next(error);
+  }
 });
 
 router.delete("/:contactId", async (req, res, next) => {
-  res.json({ message: "template message" });
+  try {
+    const { contactId } = req.params;
+    const result = await contactsOperations.removeContact(contactId);
+    if (!result) {
+      throw createError(404, "Not found");
+    }
+    res.status(200).json({
+      status: "success",
+      code: 200,
+      message: "contact deleted",
+      data: {
+        result,
+      },
+    });
+  } catch (error) {
+    next(error);
+  }
 });
 
 router.put("/:contactId", async (req, res, next) => {
-  res.json({ message: "template message" });
+  try {
+    const { error } = schemaUpdate.validate(req.body);
+    if (error) {
+      throw createError(400, "missing fields");
+    }
+    const { contactId } = req.params;
+    const result = await contactsOperations.updateContact(contactId, req.body);
+    if (!result) {
+      throw createError(404, "Not found");
```

Comment:
```suggestion
      return next(createError(404, "Not found"));
```
Tu nie robisz throw, tylko return next albo return res.status(400).send